src/plugins/intel_cpu/src/nodes/reshape.cpp

```diff
@@ -121,14 +127,28 @@ void Reshape::initSupportedPrimitiveDescriptors() {
         config.inConfs[i].setMemDesc(creatorsMap.at(LayoutType::ncsp)->createSharedDesc((i > 0 ? secondInPrc : inPrec), getInputShapeAtPort(i)));
     }
     config.outConfs.resize(1);
-    config.outConfs[0].inPlace(0);
+    config.outConfs[0].inPlace(canBeInPlace ? 0 : -1);
     config.outConfs[0].constant(false);
     config.outConfs[0].setMemDesc(creatorsMap.at(LayoutType::ncsp)->createSharedDesc(outPrec, getOutputShapeAtPort(0)));
     supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown);
 }
 
 void Reshape::executeDynamicImpl(mkldnn::stream strm) {
-    execute(strm);
+    auto& srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+
+    auto srcPtr = static_cast<uint8_t*>(srcMemPtr->GetPtr());
+    auto dstPtr = static_cast<uint8_t*>(dstMemPtr->GetPtr());
+
+    if (dstPtr != srcPtr) {
+        cpu_memcpy(dstPtr, srcPtr, dstMemPtr->GetSize());
+    }
```

Comment:
To generalize this solution, please move this implementation into `execute()` method and just call it inside `executeDynamicImpl`. Because in this implementation, it is technically possible that in the static shape case not inPlace mode is selected, but the execute call will do nosing.

Reply:
But isExecutable() can only bypass executeDynamicImpl().
If we do it in execute(), it will always introduce overhead to static shape cases(which is exactly we are trying to avoid).

Reply:
Please see https://github.com/openvinotoolkit/openvino/blob/ed8c9d6f9ac76e82081bba65b6d3d38a141c1cd9/src/plugins/intel_cpu/src/graph.cpp#L454
and
https://github.com/openvinotoolkit/openvino/blob/ed8c9d6f9ac76e82081bba65b6d3d38a141c1cd9/src/plugins/intel_cpu/src/graph.cpp#L963
If the debug caps are disabled, `isExecutable()==false` nodes will not be added to the `executableGraphNodes` list, and `execute` will not be called on them.

Reply:
OK, I see, Thank you, will change!